src/main/java/pl/javastart/bootcamp/lekcia21zadaniespringmvcprodukty/ProductController.java

```diff
@@ -0,0 +1,59 @@
+package pl.javastart.bootcamp.lekcia21zadaniespringmvcprodukty;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+
+@Controller()
+public class ProductController {
+
+    private ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String home() {
+        return "home";
+    }
+
+    @GetMapping("/lista")
+    public String list(Model model, @RequestParam(required = false, name = "kategoria") Category category) {
+        prepareDataForList(model, category);
+        return "list";
+    }
+
+    private void prepareDataForList(Model model, Category category) {
+        List<Product> products;
+
+        if (category != null) {
+            products = productRepository.findByCategory(category);
+        } else {
+            products = productRepository.findAll();
+        }
+
+        double sum = products.stream()
+                .mapToDouble(Product::getPrice).sum();
```

Comment:
`sum()` w nowej linii

Reply:
Ogólnie ładniej by wyglądało jakby tylko to by było wyniesione do nowej metody np `sumPrices(List<Product> products)` a cała logika z `prepareDataForList` była wrzucona do metody `list` controllera

Reply:
Ale nie jest to błąd

Reply:
Poprawione